tasks/eva_3417/src/main/groovy/eva3417/RemediateLowerCaseRefAlt.groovy

```diff
@@ -0,0 +1,270 @@
+        List<String> impactedSVEIds = impactedSVEList.stream()
+                .map { sve -> sve.getHashedMessage() }
+                .collect(Collectors.toList())
+        Criteria filterCriteria = new Criteria("inactiveObjects.hashedMessage").in(impactedSVEIds)
+                .and("eventType").is("UPDATED")
+        new RetryableBatchingCursor<SubmittedVariantOperationEntity>(filterCriteria, dbEnv.mongoTemplate, svoeClass).each {
+            sveOperationBatchList ->
+                {
+                    def svoeBulkUpdates = dbEnv.mongoTemplate.bulkOps(BulkOperations.BulkMode.UNORDERED, svoeClass)
+
+                    for (SubmittedVariantOperationEntity svoe : sveOperationBatchList) {
+                        // create update statement
+                        SubmittedVariantInactiveEntity inactiveSVEDocument = svoe.getInactiveObjects()[0]
+                        String oldHash = inactiveSVEDocument.getHashedMessage()
+                        String newRef = inactiveSVEDocument.getReferenceAllele().toUpperCase()
+                        String newAlt = inactiveSVEDocument.getAlternateAllele().toUpperCase()
+                        String newHash = mapOldHashNewHash.get(oldHash)
+
+                        Query findQuery = query(where("inactiveObjects.hashedMessage").is(oldHash))
+                        Update updateQuery = new Update()
+                                .set("inactiveObjects.\$.ref", newRef)
+                                .set("inactiveObjects.\$.alt", newAlt)
+                                .set("inactiveObjects.\$.hashedMessage", newHash)
+
+                        svoeBulkUpdates.updateOne(findQuery, updateQuery)
+                    }
+
+                    //execute all bulk updates
+                    svoeBulkUpdates.execute()
+                }
+        }
+
+    }
+
+    void remediate() {
+        // read all the documents from SubmittedVariantEntity
+        new RetryableBatchingCursor<SubmittedVariantEntity>(new Criteria(), dbEnv.mongoTemplate, sveClass).each {
+            sveBatchList ->
+                {
+                    // find which sve from the batch are impacted by lower case nucleotide issue
+                    List<SubmittedVariantEntity> impactedSVEList = getImpactedSVEList(sveBatchList)
+
+                    // if there are any sve impacted
+                    if (!impactedSVEList.isEmpty()) {
+                        // make a map of old hash ids of sve and new hash
+                        Map<String, String> mapOldHashNewHash = impactedSVEList.stream()
+                                .collect(Collectors.toMap(sve -> sve.getHashedMessage(), sve -> getSVENewHash(sve)))
+
+                        // find which of the new hashes already exist in db (will cause hash collision)
+                        List<String> newHashExistList = getExistingHashList(mapOldHashNewHash.values())
```

Comment:
Suggest making this a set with toSet() so that the "contains()" below is more efficient....

Reply:
done